packages/react/src/components/Toggletip/index.tsx

```diff
@@ -173,11 +174,31 @@ export function Toggletip<E extends ElementType = 'span'>({
     }
   });
 
-  useWindowEvent('click', ({ target }) => {
-    if (open && target instanceof Node && !ref.current?.contains(target)) {
-      actions.close();
-    }
-  });
+  useEffect(() => {
+    if (!open || !ref.current) return;
+
+    const targetDocument = ref.current.ownerDocument || document;
+    const eventType: 'pointerdown' | 'mousedown' =
+      'PointerEvent' in window ? 'pointerdown' : 'mousedown';
+
+    const handleOutsideClick = (event: MouseEvent | PointerEvent) => {
+      const node = event.target as Node | null;
+      if (node && !ref.current!.contains(node)) {
+        setOpen(false);
+      }
+    };
```

Comment:
```suggestion
  useEffect(() => {
    if (!ref.current) return;

    const targetDocument = ref.current.ownerDocument || document;
    const eventType: 'pointerdown' | 'mousedown' =
      'PointerEvent' in window ? 'pointerdown' : 'mousedown';

    const handleOutsideClick = (event: MouseEvent | PointerEvent) => {
      const node = event.target as Node | null;
      if (open && node && !ref.current!.contains(node)) {
        setOpen(false);
      }
    };
```

Really small thing - it seems like this event handler would get removed and re-added every time `open` changes. Can we just let the handler check `open`?

Reply:
Good point, thanks @kennylam! I updated it to check open inside the handler instead.